### ISSUE TITLE
rds_param_group module not able to set params

### ISSUE THREAD

**Issue by conlon** [ansible/ansible#24340]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

rds_param_group

##### ANSIBLE VERSION

```
ansible 2.3.0.0 (detached HEAD d56ba0945d) last updated 2017/04/14 10:38:39 (GMT -500)
  config file = /Users/mconlon/git/hyperwave_relay_jane/ansible.cfg
  configured module search path = [u'custom_modules']
  python version = 2.7.12 (default, Oct 11 2016, 05:20:59) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.38)]
```

##### CONFIGURATION

ansible.cfg:
```
[defaults]
roles_path = roles
lookup_plugins = lookup_plugins
filter_plugins = filter_plugins
hash_behaviour = merge
library = custom_modules
forks = 50

[ssh_connection]
scp_if_ssh = True
pipelining = True
# an attempt to fix ssh unreachable errors when machines are actually reachable
retries = 5
```

##### OS / ENVIRONMENT

Mac OS 10.12.3

##### SUMMARY

The module can create an rds_param_group but it can't set params

##### STEPS TO REPRODUCE

this creates a param group:
```yaml
- name: create rds parameter group(s)
      rds_param_group:
        region: "{{ region }}"
        name: "test-param-group"
        description: "test-param-group"
        engine: postgres9.5
        state: present
```

this throws a ValueError:
```yaml
- name: create rds parameter group(s)
      rds_param_group:
        region: "{{ region }}"
        name: "test-param-group"
        description: "test-param-group"
        params:
          random_page_cost: 1
        engine: postgres9.5
        state: present
```

this throws a different ValueError:
```yaml
- name: create rds parameter group(s)
      rds_param_group:
        region: "{{ region }}"
        name: "test-param-group"
        description: "test-param-group"
        params:
          log_min_duration_statement: 50
        engine: postgres9.5
        state: present
```

##### EXPECTED RESULTS

Params should get set on a param group.

##### ACTUAL RESULTS

random_page_cost test case:
```
Using module file /Users/mconlon/git/hyperwave_relay_jane/ansible2300/src/ansible/lib/ansible/modules/cloud/amazon/rds_param_group.py
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: mconlon
<localhost> EXEC /bin/sh -c '/usr/bin/python && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_aaV9p1/ansible_module_rds_param_group.py", line 316, in <module>
    main()
  File "/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_aaV9p1/ansible_module_rds_param_group.py", line 292, in main
    changed_params, group_params = modify_group(next_group, group_params, immediate)
  File "/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_aaV9p1/ansible_module_rds_param_group.py", line 213, in modify_group
    set_parameter(param, new_value, immediate)
  File "/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_aaV9p1/ansible_module_rds_param_group.py", line 182, in set_parameter
    param.value = converted_value
  File "/usr/local/lib/python2.7/site-packages/boto/rds/parametergroup.py", line 169, in set_value
    self._set_string_value(value)
  File "/usr/local/lib/python2.7/site-packages/boto/rds/parametergroup.py", line 141, in _set_string_value
    raise ValueError('value must be in %s' % self.allowed_values)
ValueError: value must be in 0-1.79769e+308

fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "module_stderr": "Traceback (most recent call last):\n  File \"/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_aaV9p1/ansible_module_rds_param_group.py\", line 316, in <module>\n    main()\n  File \"/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_aaV9p1/ansible_module_rds_param_group.py\", line 292, in main\n    changed_params, group_params = modify_group(next_group, group_params, immediate)\n  File \"/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_aaV9p1/ansible_module_rds_param_group.py\", line 213, in modify_group\n    set_parameter(param, new_value, immediate)\n  File \"/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_aaV9p1/ansible_module_rds_param_group.py\", line 182, in set_parameter\n    param.value = converted_value\n  File \"/usr/local/lib/python2.7/site-packages/boto/rds/parametergroup.py\", line 169, in set_value\n    self._set_string_value(value)\n  File \"/usr/local/lib/python2.7/site-packages/boto/rds/parametergroup.py\", line 141, in _set_string_value\n    raise ValueError('value must be in %s' % self.allowed_values)\nValueError: value must be in 0-1.79769e+308\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 1
}
```

log_min_duration_statement test case:
```
Using module file /Users/mconlon/git/hyperwave_relay_jane/ansible2300/src/ansible/lib/ansible/modules/cloud/amazon/rds_param_group.py
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: mconlon
<localhost> EXEC /bin/sh -c '/usr/bin/python && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_0eQg2e/ansible_module_rds_param_group.py", line 316, in <module>
    main()
  File "/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_0eQg2e/ansible_module_rds_param_group.py", line 292, in main
    changed_params, group_params = modify_group(next_group, group_params, immediate)
  File "/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_0eQg2e/ansible_module_rds_param_group.py", line 213, in modify_group
    set_parameter(param, new_value, immediate)
  File "/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_0eQg2e/ansible_module_rds_param_group.py", line 182, in set_parameter
    param.value = converted_value
  File "/usr/local/lib/python2.7/site-packages/boto/rds/parametergroup.py", line 171, in set_value
    self._set_integer_value(value)
  File "/usr/local/lib/python2.7/site-packages/boto/rds/parametergroup.py", line 149, in _set_integer_value
    min, max = self.allowed_values.split('-')
ValueError: too many values to unpack

fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "module_stderr": "Traceback (most recent call last):\n  File \"/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_0eQg2e/ansible_module_rds_param_group.py\", line 316, in <module>\n    main()\n  File \"/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_0eQg2e/ansible_module_rds_param_group.py\", line 292, in main\n    changed_params, group_params = modify_group(next_group, group_params, immediate)\n  File \"/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_0eQg2e/ansible_module_rds_param_group.py\", line 213, in modify_group\n    set_parameter(param, new_value, immediate)\n  File \"/var/folders/dx/xqv6cb4n7ys7mrfpvy4k3f5h0000gn/T/ansible_0eQg2e/ansible_module_rds_param_group.py\", line 182, in set_parameter\n    param.value = converted_value\n  File \"/usr/local/lib/python2.7/site-packages/boto/rds/parametergroup.py\", line 171, in set_value\n    self._set_integer_value(value)\n  File \"/usr/local/lib/python2.7/site-packages/boto/rds/parametergroup.py\", line 149, in _set_integer_value\n    min, max = self.allowed_values.split('-')\nValueError: too many values to unpack\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 1
}
```

**Comment by ansibot** [comment]
cc @tastychutney @scottanderson42
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by s-hertel** [comment]
Hi @conlon! Thanks for opening this issue. Could you test my patch please and give me some feedback if it doesn't work as expected? #24374

**Comment by conlon** [comment]
@s-hertel your patch definitely allows me to edit parameters! Thank you.

My only feedback is that each time I run the module, it modifies whatever params are listed, but isn't concerned with any previously set params. In my mind, ideally it would reset all params to default except the ones I list, but I'm not sure that's necessarily best or expected.
I can certainly work with it in its current state, and it seems like addressing this feedback should be handled in another issue & PR. Just wanted to bring it up in case it warranted some discussion.

**Comment by conlon** [comment]
Oh, one more thing...it looks like it always has a result of changed? Running it multiple times without changing the params keeps showing as changed for me.

**Comment by s-hertel** [comment]
+1 thanks for the feedback, will add those fixes!

**Comment by s-hertel** [comment]
@conlon Yeah, let's handle reseting all params to default in another PR. Good idea. We could add a new option like `purge_params` or something along those lines. I've fixed it so it only modifies parameters if the new value is different than the current value and that it only shows changed: true and "new" and "old" if something has been modified. The one issue I've noticed now is that parameters like cpu_index_tuple_cost or autovacuum_analyze_scale_factor which expect a float cannot be modified. I believe this is a boto bug (it wrongly thinks the parameter type for float parameters is a string) but I need to look into it more.
